### PR TITLE
/accept with no userid accepts sole challenges

### PR DESCRIPTION
**Pull request by Morfent**
Merge this before the server pullreq.

### CODE REVIEW

File: js/client-chat.js [hunk 1]
```diff
@@ -420,6 +420,43 @@
 				challenge(targets);
 				return false;
 
+			case 'accept':
+				if (toId(target)) return text;
```
Comment (Zarel) [inline js/client-chat.js]: You should still send the team even if you specify the user...

File: js/client-chat.js [hunk 2]
```diff
@@ -420,6 +420,43 @@
...
+					this.parseCommand('/help reject');
+					return false;
+				}
+				if ($challenges.length > 1) {
+					app.addPopupMessage('You need to specify a user if you have more than one pending challenge to reject.');
+					this.parseCommand('/help reject');
+					return false;
+				}
+
+				return text;
```
Comment (Zarel) [inline js/client-chat.js]: You might as well specify the username you're rejecting from here, then the server doesn't need any changes.

File: js/client-chat.js [hunk 3]
```diff
@@ -420,6 +420,43 @@
...
+					return false;
+				}
+				if ($challenges.length > 1) {
+					app.addPopupMessage('You need to specify a user if you have more than one pending challenge to accept.');
+					this.parseCommand('/help accept');
+					return false;
+				}
+
+				var format = $('button[name="format"]', $challenges[0]).val();
+				if (window.BattleFormats[format].team === 'preset' || Object.keys(Storage.teams).length) return text;
```
Comment (Zarel) [inline js/client-chat.js]: You still need to actually send the team. :|

You can just simulate a click on the accept button if you don't want to go through all that work. In fact, that's in general probably better for  you.

Comment (Morfent) [inline js/client-chat.js]: Simulating a click with both commands would also handle the HTML of the challenge box properly, which I hadn't caught earlier.

**Comment by Morfent** [comment]
The commands now simulate clicking the accept/reject button.

File: js/client-chat.js [hunk 4]
```diff
@@ -747,6 +778,14 @@
 					this.add('/challenge - Open a prompt to challenge a user to a battle.');
 					this.add('/challenge [user] - Challenge the user [user] to a battle.');
 					return false;
+				case 'accept':
+					this.add('/accept - Accept a challenge if only one is pending.');
+					this.add('/accept [user] - Accept a challenge from the specified user.');
```
Comment (Zarel) [inline js/client-chat.js]: You still haven't implemented this part.

This still doesn't send the team etc.

Comment (Morfent) [inline js/client-chat.js]: Is it supposed to send the team even if the command fails?

Comment (Zarel) [inline js/client-chat.js]: You made it so `/accept` sends the team.

You haven't made it so `/accept [user]` sends the team.

Comment (Morfent) [inline js/client-chat.js]: Oh. It might help if ``/accept [user]`` actually worked. I really need to get better with testing

**Comment by Morfent** [comment]
``/accept [user]`` works properly now.

File: js/client-chat.js [hunk 5]
```diff
@@ -420,6 +420,58 @@
 				challenge(targets);
 				return false;
 
+			case 'accept':
+				var userid = toId(target);
+				if (userid) {
+					var $challenge = $('.pm-window').filter('div[data-userid="' + userid + '"]').find('button[name="acceptChallenge"]');
+					if (!$challenge.length) {
+						app.addPopupMessage('You do not have any pending challenge from ' + target + ' to accept.');
```
Comment (Zarel) [inline js/client-chat.js]: Should it really be a popup message? I think inline would work better.

File: js/client-chat.js [hunk 6]
```diff
@@ -420,6 +420,58 @@
...
+						return false;
+					}
+					$challenge[0].click();
+					return false;
+				}
+
+				var $challenges = $('.challenge').find('button[name=acceptChallenge]');
+				if (!$challenges.length) {
+					app.addPopupMessage('You do not have any pending challenges to accept.');
```
Comment (Zarel) [inline js/client-chat.js]: Ditto your other popup messages.

Popup messages in general are really bad UI; they should in general be replaced with other things.

Comment (Morfent) [inline js/client-chat.js]: If that's the case, should the popup message in ``acceptChallenge`` also be made inline? I only made these as popups to follow its style.

**Comment by Morfent** [comment]
Changed the popups to regular responses.

**Pull request merged**